spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigReloadAutoConfiguration.java

```diff
@@ -43,10 +44,10 @@
  */
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnKubernetesAndConfigEnabled
-@ConditionalOnClass(EndpointAutoConfiguration.class)
+@ConditionalOnKubernetesReloadEnabled
+@ConditionalOnClass({ EndpointAutoConfiguration.class, RestartEndpoint.class, ContextRefresher.class })
```

Comment:
these two : `RestartEndpoint.class, ContextRefresher.class` where part of `@ConditionalOnClass`, so I just moved them in here, since we dropped the inner class. Initially I put them on `@AutoConfigureAfter`, but this was a brain fart from my side, as I placed them under the wrong annotation. sorry about that